src/promptflow/promptflow/_core/tools_manager.py

```diff
@@ -194,23 +200,54 @@ def append_workspace_triple_to_func_input_params(func_sig_params, func_input_par
     return combined_func_input_params
 
 
-def gen_dynamic_list(func_path: str, func_input_params_dict: Dict, ws_triple_dict: Dict[str, str] = None):
-    import importlib
-    import inspect
+def _validate_response_type(response: Any, f: str):
+    """Verify response type is correct.
+
+    The response is a list of items. Each item is a dict with the following keys:
+        - value: for backend use. Required.
+        - display_value: for UI display. Optional.
+        - hyperlink: external link. Optional.
+        - description: information icon tip. Optional.
+    The response can not be empty.
+    """
+    if not response:
+        raise ListFunctionResponseError(f"{f} response can not be empty.")
```

Comment:
Let's improve the message to be more friendly to tool user who may be unfamiliar with the tool impl.

Reply:
For example, "cannot list items due to xxx. Please contact tool author for trouble shooting."

Reply:
added